lib/kits/core-ui/paged-publication.js

```diff
@@ -0,0 +1,206 @@
+import * as PagedPublicationKit from '../paged-publication';
+import * as PagedPublicationComponents from './components/paged-publication';
+import * as clientLocalStorage from '../../storage/client-local';
+import {request} from '../core';
+import SGN from '../../sgn-sdk';
+
+const pagedPublication = (scriptEl, mainContainer = '') => {
+    let options = {};
+    let sgnData = {};
+    let sgnViewer = null;
+    const scriptEls = {
+        businessId: scriptEl.getAttribute('data-business-id'),
+        mainContainer:
+            scriptEl.getAttribute(
+                'data-component-paged-publication-container'
+            ) || mainContainer,
+        publicationId: scriptEl.getAttribute('data-component-publication-id'),
+        publicationIdParam:
+            scriptEl.getAttribute('data-publication-id-query-param') ||
+            'publicationid',
+        pageIdParam:
+            scriptEl.getAttribute('data-publication-page-query-param') ||
+            'page',
+        localeCode: scriptEl.getAttribute('data-locale-code'),
+        translationKeys: scriptEl.getAttribute(
+            'data-translation-key-paged_publication_viewer'
+        ),
+        removedComponents: scriptEl.getAttribute(
+            'data-component-paged-publication-remove-components'
+        ),
+        theme: scriptEl.getAttribute('data-component-theme')
+    };
+
+    const customTemplates = {
+        mainContainer: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-template'
+        ),
+        offerList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-offer-list-template'
+        ),
+        shoppingList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-shopping-list-template'
+        ),
+        shoppingListCounter: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-shopping-list-counter-template'
+        ),
+        pageList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-page-list-template'
+        ),
+        menuContainer: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-menu-container-template'
+        )
+    };
+
+    const renderMainContainer = () => {
+        const mainContainerEl = document.querySelector(scriptEls.mainContainer);
+
+        PagedPublicationComponents.MainContainer({
+            template: customTemplates.mainContainer,
+            shoppingListCounterTemplate: customTemplates.shoppingListCounter,
+            el: mainContainerEl,
+            scriptEls
+        }).render();
+    };
+    renderMainContainer();
+
+    const pagedPublicationObj = {
+        render: async () => {
+            if (Object.keys(options).length === 0) {
```

Comment:
I think almost every method on this object can be a free function instead, from what I can see only `render` is used by `pagedPublication` callers - we don't need to expose all these other functions to consumers(or risk them not being code eliminated if unused).